Trading/Mode/arbitrage_trading_mode/arbitrage_trading_mode.py

```diff
@@ -0,0 +1,461 @@
+        except Exception as e:
+            self.logger.exception(e, True, f"Error when handling mark_price_callback for {self.exchange_name}: {e}")
+
+    async def _analyse_arbitrage_opportunities(self):
+        other_exchanges_average_price = mean(self.other_exchanges_mark_prices.values())
+        state = None
+        if other_exchanges_average_price > self.own_exchange_mark_price / self.triggering_price_delta_ratio:
+            # min long = high price > own_price / (1 - 2fees)
+            state = EvaluatorStates.LONG
+        elif other_exchanges_average_price < self.own_exchange_mark_price * self.triggering_price_delta_ratio:
+            # min short = low price < own_price * (1 - 2fees)
+            state = EvaluatorStates.SHORT
+        if state is not None:
+            # lock to prevent concurrent order management
+            async with self.lock:
+                # 1. cancel invalided opportunities if any
+                await self._ensure_no_expired_opportunities(other_exchanges_average_price, state)
+                # 2. handle new opportunities
+                await self._trigger_arbitrage_opportunity(other_exchanges_average_price, state)
+
+    async def _trigger_arbitrage_opportunity(self, other_exchanges_average_price, state):
+        # ensure no similar arbitrage is already in place
+        if self._ensure_no_existing_arbitrage_on_this_price(state):
+            self.logger.debug(f"Arbitrage opportunity on {self.exchange_manager.exchange_name} {state.name} for "
+                              f"{self.trading_mode.symbol} "
+                              f"({self.own_exchange_mark_price} vs {other_exchanges_average_price} on average)")
+            arbitrage_container = ArbitrageContainer(self.own_exchange_mark_price, other_exchanges_average_price, state)
+            await self._create_arbitrage_initial_order(arbitrage_container)
+            self._register_state(state, other_exchanges_average_price - self.own_exchange_mark_price)
+
+    async def _create_arbitrage_initial_order(self, arbitrage_container):
+        if self.exchange_manager.trader.is_enabled:
+            data = {
+                ArbitrageModeConsumer.ARBITRAGE_CONTAINER_KEY: arbitrage_container,
+                ArbitrageModeConsumer.ARBITRAGE_PHASE_KEY: ArbitrageModeConsumer.INITIAL_PHASE
+            }
+            await self.submit_trading_evaluation(cryptocurrency=self.trading_mode.cryptocurrency,
+                                                 symbol=self.trading_mode.symbol,
+                                                 time_frame=None,
+                                                 state=arbitrage_container.state,
+                                                 data=data)
+
+    async def _trigger_arbitrage_secondary_order(self, arbitrage, filled_order, filled_quantity_before_fees):
+        arbitrage.passed_initial_order = True
+        now_buying = arbitrage.state is EvaluatorStates.SHORT
```

Comment:
`is not EvaluatorStates.SHORT` ?

Reply:
in this context an ```EvaluatorStates.SHORT``` arbitrage is when you first sell to buy later, in this case, the secondary order is a buy

Reply:
Oh ok, can you explain it with a comment above please ?